pkg/reaper/deployment_test.go

```diff
@@ -33,8 +35,20 @@ func TestNewDeployment(t *testing.T) {
 			Name: "truststore-secret",
 		}},
 	}
+	reaper.Spec.ResourceMeta = &meta.ResourceMeta{
```

Comment:
Could you add a test case that verifies that a label present in both `CommonLabels` and `Pods.Labels`  gets correctly merged? Same for `CommonLabels` + `Service.Labels`.

Reply:
with [this ](https://github.com/k8ssandra/k8ssandra-operator/pull/788#discussion_r1053154792) change, we'll delegate adding the common labels to cass-operator. Added some `commonLabel` fields in existing tests to show that.

Reply:
In `CassandraDatacenter` resources, yes, but not for `Reaper` and `Stargate` resources: for these, we do the overlay. The current test design is fine, as it shows that deployment will have `override -> commonLevel` but pods will have `override -> podLevel`.